### DISCUSSION TITLE
Using AIR as a Library within native apps

### DISCUSSION THREAD

**Discussion by yourfungames** [airsdk/Adobe-Runtime-Support#2060] [General]
Hi,
Just found that Unity had launched a feature (in 2020) - Using unity as a library within native apps (ios, android, flutter)  - https://www.youtube.com/watch?v=2zQYrOeF1Ko
Can we do something similar with AIR? 

Regards,
nis

**Comment by marchbold** [comment]
Flutter have a similar feature for developers migrating part of an application across, (https://docs.flutter.dev/development/add-to-app) I created a wrapper to get flutter running in an AIR app which wasn't too hard. 

Would be an interesting concept to explore for AIR.

**Reply by ajwfrost** [reply]
Yes, we've looked a little at this (mostly with a view to working around some of the macOS permissions/restrictions) - we just need to make sure we can still fit this all within the licensing arrangements for AIR apps, i.e. so that people are covered by the "distribution" permissions for the AIR runtimes.

**Reply by marchbold** [reply]
Ooo that sounds exciting :)
